react-ui/src/utils/mathjs/derivatives.js

```diff
@@ -23,21 +24,49 @@ export const diff = (f: NumericFunction, ...values: [] | Array<Numeric>) => {
   }
   Object.defineProperty(derivative, 'length', { value: f.length } )
 
+  // $FlowFixMe
   return values.length === 0
     ? derivative
     : derivative(...values)
 }
 
-type Func3to1 = (t: number) => [number, number, number]
-export const unitT = (f: Func3to1, t: number) => {
+type Func1to3 = (t: number) => [number, number, number]
+
+export const unitT = (f: Func1to3, t: number) => {
   // $FlowFixMe ... need to make add/subtract polymorhpic & length-preserving
-  const tangent = math.subtract(f(t + EPS/2), f(t - EPS/2))
+  const tangent = math.subtract(f(t + EPS2), f(t - EPS2))
   return math.divide(tangent, math.norm(tangent))
 }
-export const unitN = (f: Func3to1, t: number) => {
-  const normal = math.subtract(unitT(f, t + EPS/2), unitT(f, t - EPS/2))
+export const unitN = (f: Func1to3, t: number) => {
+  const normal = math.subtract(unitT(f, t + EPS2), unitT(f, t - EPS2))
   return math.divide(normal, math.norm(normal))
 }
-export const unitB = (f: Func3to1, t: number) => {
+export const unitB = (f: Func1to3, t: number) => {
   return math.cross(unitT(f, t), unitN(f, t))
 }
+
+export const curl = (f: NumericFunction, x: number, y: number, z: number) => {
+  const [dxf, dyf, dzf] = diff(f, x, y, z)
+  // $FlowFixMe flow has trouble recognizing that diff() returned a 3 x 3 matrix
+  return [ dyf[2] - dzf[1], dzf[0] - dxf[2], dxf[1] - dyf[0] ]
+}
+
+export const div = (f: NumericFunction, x: number, y: number, z: number) => {
```

Comment:
`div` here is defined only for functions with 3 parameters, but the changelog implies any number of parameters.

Reply:
Good catch. BTW: none of these functions give particularly nice error hints at the moment, but that's true of all of them (diff, unitT, etc) not just the new ones (div, curl, pdiff) so I may work on that in a separate PR.

Sorry for the slowness in replies this past week, I've been on vacation (and am, still, so probably won't be around too much). But I fixed the issue you brought up and am going to push this to the main site now.